ARGO_Team_D/ARGO_Team_D/ControlSystem.cpp

```diff
@@ -2,63 +2,60 @@
 
 ControlSystem::ControlSystem()
 {
-
 }
 
 ControlSystem::~ControlSystem()
 {
 }
 
-void ControlSystem::handleInput(SDL_Keycode key)
+void ControlSystem::update()
 {
-	/*for (int i = 0; i < m_entityList.size(); i++)
+	std::vector<string> allowedTypes = { "Body" };
+	for (auto &e : m_entityList)
 	{
-		ControlComponent * contComp = dynamic_cast<ControlComponent *>(m_entityList.at(i));
-		if (contComp != nullptr)
+		auto comps = e->getComponentsOfType(allowedTypes);
+		BodyComponent * bodyComp = dynamic_cast<BodyComponent *>(comps["Body"]);
+		if (bodyComp != nullptr)
```

Comment:
From these lines on until 39 will only ever get execute once per loop if I'm right

Reply:
So once per loop we'd iterate over everything on this but only ever apply an effect once. Seems like they should be more command related stuff to me but not sure what the intention is atm

Reply:
Discussed an will review later